reana_server/rest/launch.py

```diff
@@ -147,7 +152,10 @@ def launch(user, url, name="", parameters="{}", specification=None):
 
         # Load and validate the workflow spec
         spec_path = fetcher.workflow_spec_path()
-        reana_yaml = load_reana_spec(spec_path, workspace_path=tmpdir)
+        # FIXME: locking will not be needed when the loading and validation of
+        # specifications will be done inside an external sandbox
+        with load_reana_spec_lock:
```

Comment:
Could this lead to situations with lock being stalled and never released, which could block further requests? I guess we can simply restart `r-server` pod when this happens, which is acceptable... but I was wondering whether we could do some gentle "auto-release-when-stalled" sort of handling here? (Not necessary to implement for the merge.)

Reply:
> Could this lead to situations with lock being stalled and never released, which could block further requests?

Yes, it might happen if `load_reana_spec` stalls or takes a long time to complete.  Putting a limit on the running time of `load_spec_reana` is for sure a good idea and it should be easy to do when we will load the specification in an external sandbox. Right now, I don't think we have any mechanism to control how much time `load_reana_spec` takes.